BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -139,7 +142,14 @@ - (void)viewDidLoad
 {
     [super viewDidLoad];
 
-    [self loadAuthorizationURL];
+    if ([self.SDKClient.session isKindOfClass:[BOXOAuth2Session class]]) {
+        [self loadAuthorizationURL];
+    } else if ([self.SDKClient.session isKindOfClass:[BOXAppUserSession class]]) {
+        __weak BOXAuthorizationViewController *me = self;
+        [self.SDKClient autheticateAppUserWithCompletionBlock:^(BOXUser *user, NSError *error) {
```

Comment:
If we're using a BOXAppUserSession, don't we know for a fact that this is what will get called by ' - (void)authenticateWithCompletionBlock:(void (^)(BOXUser *user, NSError *error))completion;' ?